### PR TITLE
build (travis): add execution of tests on Travis-CI

### PR DESCRIPTION
**Pull request by subotic**
### Summary

 - Sipi tests are executed on Travis-CI
 - A docker image is built and published

### CODE REVIEW

**Comment by benjamingeer** [comment]
I guess I should start working again on making more meaningful tests. :)

**Comment by lrosenth** [comment]
By the way, SIPI is a great hit here in Rome!! Lot’s of interest from many institutions:
- Getty in LA
- Harvard Library

:-) We’re on a very good track ;-) so more tests is nice ;-)

Am 09.06.2017 um 14:28 schrieb Ivan Subotic <notifications@github.com<mailto:notifications@github.com>>:

Summary

  *   Sipi tests are executed on Travis-CI
  *   A docker image is built and published

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/dhlab-basel/Sipi/pull/164

Commit Summary

  *   build (travis): add running tests
  *   build (travis): add running tests
  *   build (travis): add running tests
  *   build (travis): add running tests
  *   build (travis): add running tests
  *   build (travis): add building and uploading of image to docker hub
  *   build (travis): add sipi docker config with absolute paths
  *   build (travis): add sipi docker config with absolute paths
  *   build (travis): add building and uploading of image to docker hub
  *   build (travis): remove slashes from docker tag
  *   build (travis): remove slashes from docker tag
  *   build (travis): change webapi hostname
  *   build (travis): add stages

File Changes

  *   M .travis.yml<https://github.com/dhlab-basel/Sipi/pull/164/files#diff-0> (60)
  *   M Dockerfile<https://github.com/dhlab-basel/Sipi/pull/164/files#diff-1> (18)
  *   A config/sipi.knora-test-docker-config.lua<https://github.com/dhlab-basel/Sipi/pull/164/files#diff-2> (175)

Patch Links:

  *   https://github.com/dhlab-basel/Sipi/pull/164.patch
  *   https://github.com/dhlab-basel/Sipi/pull/164.diff

File: config/sipi.knora-test-config.lua [hunk 1]
```diff
@@ -142,6 +142,11 @@ sipi = {
     loglevel = "DEBUG",
 }
 
+fileserver = {
+    docroot = './server',
+    docroute = '/server'
+}
```
Comment (benjamingeer) [inline config/sipi.knora-test-config.lua]: Is there really a `docroot` and a `docroute`?

Comment (subotic) [inline config/sipi.knora-test-config.lua]: Yes, this is the new fileserver feature. I agree that the naming could be a bit clearer.

Comment (retokromer) [inline config/sipi.knora-test-config.lua]: Indeed.

Comment (benjamingeer) [inline config/sipi.knora-test-config.lua]: Opened #168 for this.

**Comment by subotic** [comment]
Thanks :-)

**Pull request merged**